File: main/src/main/java/lavaplayer/source/youtube/format/YoutubeTrackFormatExtractor.java
package lavaplayer.source.youtube.format;

import lavaplayer.source.youtube.YoutubeSignatureResolver;
import lavaplayer.source.youtube.YoutubeTrackFormat;
import lavaplayer.source.youtube.YoutubeTrackJsonData;
import lavaplayer.tools.io.HttpInterface;

import java.util.List;

public interface YoutubeTrackFormatExtractor {
    String DEFAULT_SIGNATURE_KEY = "signature";

    List<YoutubeTrackFormat> extract(
        YoutubeTrackJsonData response,
        HttpInterface httpInterface,
        YoutubeSignatureResolver signatureResolver
    );
}

File: main/src/main/java/lavaplayer/source/youtube/format/YoutubeTrackFormatExtractor.kt
package lavaplayer.source.youtube.format

import lavaplayer.source.youtube.YoutubeSignatureResolver
import lavaplayer.source.youtube.YoutubeTrackFormat
import lavaplayer.source.youtube.YoutubeTrackJsonData
import lavaplayer.tools.io.HttpInterface

interface YoutubeTrackFormatExtractor {
    companion object {
        const val DEFAULT_SIGNATURE_KEY = "signature"
    }

    fun extract(
        response: YoutubeTrackJsonData,
        httpInterface: HttpInterface,
        signatureResolver: YoutubeSignatureResolver
    ): List<YoutubeTrackFormat>
}
